src/Models/Actor.php

```diff
@@ -0,0 +1,117 @@
+<?php
+declare(strict_types=1);
+
+namespace LotGD\Core\Models;
+
+use LotGD\Core\Exceptions\PermissionAlreadyExistsException;
+use LotGD\Core\Exceptions\PermissionDoesNotExistException;
+use LotGD\Core\Models\Permission;
+use LotGD\Core\Models\PermissionAssociationInterface;
+
+/**
+ * This abtract class provides functionality for user entities that crates might
+ * want to implement, such as permissions.
+ */
+abstract class Actor
+{
+    /** @var array Associations between permission-id and PermissionAssociation entity. */
+    private $_permissions = [];
```

Comment:
nit: is prefixing private vars with _ in a style guide somewhere? Is that common in PHP?